src/extractor/guidance/coordinate_extractor.cpp

```diff
@@ -119,35 +117,19 @@ util::Coordinate CoordinateExtractor::ExtractRepresentativeCoordinate(
     if (coordinates.size() <= 2)
     {
         // Here we can't check for validity, due to possible dead-ends with repeated coordinates
-        // BOOST_ASSERT(is_valid_result(coordinates.back()));
         return coordinates.back();
     }
 
     // due to repeated coordinates / smaller offset errors we skip over the very first parts of the
     // coordinate set to add a small level of fault tolerance
     const constexpr double skipping_inaccuracies_distance = 2;
 
-    // fallback, mostly necessary for dead ends
-    if (intersection_node == to_node)
```

Comment:
removing this will actually negatively affect the turn discovery. It has nothing to do with the changes necessary and has a negative impact on discovery in traffic loops.